### PR TITLE
Facial hair now checks for a bodytype, rather than pronouns

### PR DESCRIPTION
**Pull request by quardbreak**

### CODE REVIEW

File: code/modules/client/preference_setup/general/01_basic.dm [hunk 1]
```diff
@@ -105,6 +103,8 @@
 			pref.bodytype = new_body.name
 			if(new_body.associated_gender) // Set to default for male/female to avoid confusing people
 				pref.gender = new_body.associated_gender
+			if(!(pref.f_style in S.get_facial_hair_styles(new_body.associated_gender)))
```
Comment (MistakeNot4892) [inline code/modules/client/preference_setup/general/01_basic.dm:106]: You need to update every place where get_facial_hair_styles() and get_hair_styles() are called so they use bodytype gender. Doing it piecemeal just means there'll be issues elsewhere.

Comment (Koollan) [reply code/modules/client/preference_setup/general/01_basic.dm:106]: alternatively, we could just have facial hair not locked behind gender regardless, seems a lot simpler that way and more customizable

Comment (MistakeNot4892) [reply code/modules/client/preference_setup/general/01_basic.dm:106]: Yep, but if people later want to have markings restricted by bodytype (sexual dimorphism or variant silhouettes) the system will still be broken.

**Comment by quardbreak** [comment]
Ready for review.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**